waku/v2/node/waku_node.nim

```diff
@@ -853,59 +853,47 @@ proc startKeepalive*(node: WakuNode) =
   asyncSpawn node.keepaliveLoop(defaultKeepalive)
 
 proc runDiscv5Loop(node: WakuNode) {.async.} =
-  ## Continuously add newly discovered nodes
-  ## using Node Discovery v5
-  if (node.wakuDiscv5.isNil):
+  ## Continuously add newly discovered nodes using Node Discovery v5
+  if node.wakuDiscv5.isNil():
     warn "Trying to run discovery v5 while it's disabled"
     return
 
-  info "Starting discovery loop"
+  info "starting discv5 discovery loop"
 
   while node.wakuDiscv5.listening:
-    trace "Running discovery loop"
-    let discoveredPeersRes = await node.wakuDiscv5.findRandomPeers()
+    trace "running discv5 discovery loop"
+    let discoveredRecords = await node.wakuDiscv5.findRandomPeers()
+    let discoveredPeers = discoveredRecords.mapIt(it.toRemotePeerInfo()).filterIt(it.isOk()).mapIt(it.value)
 
-    if discoveredPeersRes.isOk:
-      let discoveredPeers = discoveredPeersRes.get
-      let newSeen = discoveredPeers.countIt(not node.peerManager.peerStore[AddressBook].contains(it.peerId))
-      info "Discovered peers", discovered=discoveredPeers.len, new=newSeen
+    for peer in discoveredPeers:
+      let isNew = not node.peerManager.peerStore[AddressBook].contains(peer.peerId)
+      debug "peer discovered", peer= $peer, origin= "discv5", is_new= $isNew
```

Comment:
isnt this to much to log one line for every peer? perhaps we can trace it, but debug is too much imho.

i wouldnt remove the "Discovered peers" as its quite relevant to see if we are finding new peers. Im fine with moving it to debug if info is too verbose, but wouldnt remove it.

Reply:
> isnt this to much to log one line for every peer? perhaps we can trace it, but debug is too much imho.

Maybe. We can just log the new peers in debug level.


> i wouldnt remove the "Discovered peers" as its quite relevant to see if we are finding new peers. Im fine with moving it to debug if info is too verbose, but wouldnt remove it.

It is not about verbosity. `info` level should never be used in libraries. This goes against standard practices and, as a consequence, the expectations of the library users. The standard is that a library must not emit logs unless the `debug` level is enabled.

Typically, `info`, `warn`, and `error` are application-specific log levels. If one library uses the `error` level clearly shows that there is an API design flaw. We have `Result`s and `Exception`s to convey these "erroneous states" up in the call chain.

Based on this reasoning: `waku/v2/waku_node` is a library module. It should not use `info`, `warn`, or `error` logs. If used, those are code smells and should be converted to either `Result`/`Defect` or pushed one layer up to the application: `wakunode2/app.nim#App`.

----

Note that this is not just my opinion; it is based on facts and good practices and aligns with a discussion about logging that happened in the Nimbus server (for example).

Reply:
> > isnt this to much to log one line for every peer? perhaps we can trace it, but debug is too much imho.
> 
> Maybe. We can just log the new peers in debug level.

Changed here: https://github.com/waku-org/nwaku/compare/975924dd1b6c1e7c67aa49a6ef7a14448571af5d..b972dd88a66f7afa97410b133e9825bec817dd18

Reply:
not sure I follow.

This log has been removed, which I think is important.

`info "Discovered peers", discovered=discoveredPeers.len, new=newSeen`

If you dont want to use `info`, fine, use `debug`.

and regarding the log you have added, having `debug "peer discovered"` (which is logged for every discovered peer every discovery hearbeat) is going to add too many logs.

not a blocker though.